### PR TITLE
Guides tweaks and fix guides missing on initial load

### PR DESCRIPTION
**Pull request by pleek91**
# Description
Fixes
- Guides missing on initial load
- Guide warnings on initial load
- Guides lagging behind nodes (rubber band effect)
- Viewport date range v-model not syncing with the pixi viewport

### CODE REVIEW

File: src/components/RunGraph.vue
Comment (pleek91) [inline src/components/RunGraph.vue:1]: Makes sure the pixi viewport gets updated to reflect the date range if its updated externally

File: src/factories/guide.ts [hunk 1]
```diff
@@ -27,23 +30,30 @@ export async function guideFactory() {
   const label = inter('')
   element.addChild(label)
 
+  let scale = await waitForScale()
   let currentDate: Date | undefined
   let currentLabelFormatter: FormatDate
 
+  emitter.on('scaleUpdated', updated => scale = updated)
```
Comment (pleek91) [inline src/factories/guide.ts:37]: This keeps scales up to date without having to make the functions that depend on them async.

File: src/factories/guide.ts [hunk 2]
```diff
@@ -27,23 +30,30 @@ export async function guideFactory() {
...
+
   application.ticker.add(() => {
+    if (settings.disableGuides) {
+      return
+    }
```
Comment (pleek91) [inline src/factories/guide.ts:42]: Tracking if guides should be rendered or not externally same as the other things that can be disabled (animations, edges)

File: src/factories/guide.ts [hunk 3]
```diff
@@ -59,8 +69,7 @@ export async function guideFactory() {
     label.position.set(config.styles.guideTextLeftPadding, config.styles.guideTextTopPadding)
   }
 
-  async function updatePosition(): Promise<void> {
-    const scale = await waitForScale()
+  function updatePosition(): void {
```
Comment (pleek91) [inline src/factories/guide.ts:72]: Making this sync keeps it from lagging behind the viewport.

File: src/factories/guides.ts
Comment (pleek91) [inline src/factories/guides.ts:1]: This is reorganized and simplified so you can just call render to update the guides. The guides object is now responsible for updating the guides based on events.

**Review by brandonreid — APPROVED** [review]

File: src/objects/guides.ts
Comment (pleek91) [inline src/objects/guides.ts:1]: This object is now in charge of any re-renders that need to happen based on the viewport or layout changing.

File: src/objects/viewport.ts
Comment (pleek91) [inline src/objects/viewport.ts:1]: These changes are related to fixing the viewport v-model

**Pull request merged**